### ISSUE TITLE
Authenticating when password contains special characters

### ISSUE THREAD

**Issue by pvcnt** [bazelbuild/rules_jvm_external#368]
I am trying to use those rules with a Maven repository requiring authentication. It happens that my password has some special characters that need to get URL-encoded. If I don't encode those characters, I get an exception stating the that URL is invalid. If I encode those characters, I can't authenticate against the registry. After briefly reading the Bazel code responsible for this, it looks like there is nothing to decode those characters.

As a side question, is there a reason for .netrc not being supported outside of pinned mode? It would be much better anyway to store the password outside of the WORKSPACE file.

**Comment by jin** [comment]
> As a side question, is there a reason for .netrc not being supported outside of pinned mode? It would be much better anyway to store the password outside of the WORKSPACE file.

Yes. Non-pinned mode uses Coursier CLI for fetching and resolution, and Coursier doesn't support reading netrc. We'd need to implement an integration layer in the repository rule to manually wire up netrc parsing and passing information to Coursier CLI. 

With pinned mode, this is done automatically.
